content/blog/2020/07/2020-07-29-git-performance-improvement-phase2.adoc

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: "Git Plugin Performance Improvement Phase-2 Progress"
+tags:
+- plugins
+- git
+- platform-sig
+- developer
+- performance
+- gsoc
+- gsoc2020
+author: rishabhbudhouliya
+opengraph:
+  image: /images/post-images/2020/07-git-plugin-performance-improvement/gsoc-git-opengraph.png
+---
+
+The second phase of the link:https://www.jenkins.io/blog/2020/07/09/git-performance-improvement-phase1/[Git Plugin Performance Improvement project] has been great in terms of the progress we have achieved in implementing performance improvement insights
+derived from the phase-1 link:https://openjdk.java.net/projects/code-tools/jmh/[JMH] micro-benchmark experiments.
+
+As we all know that performance of a git fetch is highly correlated to the size of the remote repository, our task was to find the difference in performance for the two available git implementations in the link:https://github.com/jenkinsci/git-plugin[Git Plugin], "Git" and "JGit".
+
+Our major finding was that "Git" performs much better than "JGit" when it comes to a large sized repository (>100 MiB). Interestingly, "JGit" performs better than "Git" when size of the repository is less than 100 MiB. 
+
+In this phase, we were successful in coding this derived knowledge from the benchmarks into a new functionality called the 
+GitToolChooser.
+
+
+== GitToolChooser
+
+This class aims to add the functionality of recommending a git implementation on the basis of the size of a repository which has a strong correlation to the performance of git fetch (from Performance Benchmarks).
+
+It utilizes two heuristics to calculate the size:
+
+* Using cached .git dir from Multi-Branch Projects to estimate the size of a repository
+* Providing an extension point which upon implementation can use REST APIs exposed by git service providers like Github, GitLab etc to fetch the size of remote repository.
+
+Will it optimise your Jenkins instance?
```

Comment:
```suggestion
Will it optimize your Jenkins instance?
```
might be the American English speaking here... I'm ok to discard this suggestion 😄

Reply:
haha, I have replaced it with `z`. (🇺🇸)